tests/attr/test_deeplift_basic.py

```diff
@@ -71,6 +74,80 @@ def test_relu_linear_deeplift(self):
         # expected = [[[0.0, 0.0]], [[6.0, 2.0]]]
         self._deeplift_assert(model, DeepLift(model), inputs, baselines)
 
+    def test_deeplift_with_interim_output(self):
+        x1 = torch.tensor([[-10.0, 1.0, -5.0]], requires_grad=True)
+        x2 = torch.tensor([[3.0, 3.0, 1.0]], requires_grad=True)
+
+        b1 = torch.tensor([[0.0, 0.0, 0.0]], requires_grad=True)
+        b2 = torch.tensor([[0.0, 0.0, 0.0]], requires_grad=True)
+
+        inputs = (x1, x2)
+        baselines = (b1, b2)
+
+        class NewModel(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.main_model = ReLULinearDeepLiftModel()
+                self.additional = nn.Sigmoid()
+
+            def forward(self, x1, x2):
+                output = self.main_model(x1, x2)
+                return self.additional(output)
+
+        model = NewModel()
+        main_model = ReLULinearDeepLiftModel()
+
+        dl_truncated = DeepLift(model, target_output_layer=model.main_model)
+        dl_main = DeepLift(main_model)
+
+        attrs_truncated = dl_truncated.attribute(inputs, baselines)
+        attrs_main = dl_main.attribute(inputs, baselines)
+
+        # same as assertAttributeComparison, but we care about exact equalities
+        for attribution1, attribution2 in zip(attrs_truncated, attrs_main):
+            assert (attribution1 == attribution2).all()
+
+        # checks the target_output attribute was removed
+        with pytest.raises(AttributeError):
+            _ = model.main_model.target_output
+
+    def test_deeplift_shap_with_interim_output(self):
+        x1 = torch.tensor([[-10.0, 1.0, -5.0]], requires_grad=True)
+        x2 = torch.tensor([[3.0, 3.0, 1.0]], requires_grad=True)
+
+        b1 = torch.tensor([[0.0, 0.0, 0.0]], requires_grad=True)
+        b2 = torch.tensor([[0.0, 0.0, 0.0]], requires_grad=True)
+
+        inputs = (x1, x2)
+        baselines = (b1, b2)
+
+        class NewModel(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.main_model = ReLULinearDeepLiftModel()
+                self.additional = nn.Sigmoid()
+
+            def forward(self, x1, x2):
+                output = self.main_model(x1, x2)
+                return self.additional(output)
+
+        model = NewModel()
+        main_model = ReLULinearDeepLiftModel()
+
+        dl_truncated = DeepLiftShap(model, target_output_layer=model.main_model)
+        dl_main = DeepLiftShap(main_model)
+
+        attrs_truncated = dl_truncated.attribute(inputs, baselines)
+        attrs_main = dl_main.attribute(inputs, baselines)
```

Comment:
@gabrieltseng , I think your implementation is similar to `NeuronDeeplift` but we fix neuron index, In order to support your implementation, we'll probably need to allow `neuron_index=None` and compute the gradients of the layer (all neuron in that layer) with respect to the inputs.

 https://github.com/pytorch/captum/pull/142/files#diff-ac1add4f968d4c9d5fc3edd1141da856R25 

Your tests passed for me as I tried:

```
        dl_truncated = NeuronDeepLift(model, model.main_model)
        dl_main = DeepLift(main_model)

        attrs_truncated = dl_truncated.attribute(inputs, 0, baselines)
       ...
```

Reply:
Ah awesome! I think this means this PR can be closed.

Thanks!